posts/2022/batch-processing-notebooks-with-papermill.md

```diff
@@ -0,0 +1,142 @@
+---
+author: Anderson Banihirwe, Matthew Long
+date: 2022-03-14
+tags: python, jupyter, notebooks, papermill, automation
+---
+
+# Batch Processing Jupyter Notebooks with Papermill
+
+## Background and Motivation
+
+[Jupyter notebooks](https://jupyter.org/) are really good for doing the heavy lifting of data analysis by:
+
+- Allowing you to showcase your work in a single place. You can see the complete "paper trail" of what was done. This includes the code, results, visuals and the narrative accompanying your analysis.
+
+- Allowing others to easily use your work as a starting point for their own analysis. A new user can run cell by cell through the notebook to better get an understanding of the code.
+
+However, Jupyter notebooks have a few drawbacks:
+
+- Jupyter notebooks are difficult to maintain and reuse. Unlike a regular Python module that you can import and use in any Python project, users are copying and pasting snippets from each other's notebooks, and it's very easy to get out of sync.
+- Jupyter notebooks are hard to parameterize. This makes it difficult to maintain one version of the truth that can be used as a notebook template for exploring different parameters.
```

Comment:
Would it be useful for this audience to say what "parametrize" means in this context?

Reply:
I concur.  I think that a practical definition of "parameter" is needed, here, such as a statement saying that any variable defined in "parameterized" cell (see below) can be used as an input parameter for the notebook.  Or something like that.